src/visualization/recipes_makie.jl

```diff
@@ -102,8 +102,13 @@ function iplot(pd::PlotData2DTriangulated;
   flat_wire_points = Makie.@lift get_flat_points($wire_points, $z_offset)
   wire_mesh_flat = Makie.lines!(ax, flat_wire_points, color=:black)
 
+  # create a small variation in the extrema to avoid the Makie `range_step` cannot be zero error.
+  # see https://github.com/MakieOrg/Makie.jl/issues/931 for more details.
+  # the colorbar range is perturbed by 1e-5 * the magnitude of the solution.
+  scaled_extrema(x) = extrema(x) .+ 1e-5 .* maximum(abs.(extrema(x))) .* (-1, 1)
```

Comment:
```suggestion
  function scaled_extrema(x)
    ex = extrema(x) 
    return ex .+ 1e-5 .* maximum(abs.(ex)) .* (-1, 1)
  end
```
This avoids running though all memory twice.